src/components/structures/TimelinePanel.js

```diff
@@ -536,9 +536,16 @@ var TimelinePanel = React.createClass({
                 this.props.timelineSet.room.roomId,
                 this.state.readMarkerEventId,
                 lastReadEvent
-            ).catch(() => {
+            ).catch((e) => {
+                // /read_markers API is not implemented on this HS, fallback to just RR
+                if (e.errcode === 'M_UNRECOGNIZED') {
```

Comment:
what should be the behaviour for other errors? should we not set `last_rr_sent_event_id = undefined` so that we retry?